Source/WebCore/PAL/pal/text/EncodingTables.cpp

```diff
@@ -1073,16 +1072,19 @@ const std::array<std::pair<uint16_t, UChar>, 7724>& jis0208()
         ASSERT(!error);
 
         constexpr size_t range = 94;
-        uint8_t icuInput[2];
+        std::array<uint8_t, 2> icuInput;
         UChar icuOutput;
         for (size_t i = 0; i < range; i++) {
             for (size_t j = 0; j < range; j++) {
                 icuInput[0] = 0xA1 + i;
                 icuInput[1] = 0xA1 + j;
 
-                UChar* output = &icuOutput;
-                const char* input = byteCast<char>(&icuInput[0]);
-                ucnv_toUnicode(icuConverter.get(), &output, output + 1, &input, input + sizeof(icuInput), nullptr, true, &error);
+                auto outputSpan = singleElementSpan(icuOutput);
+                UChar* output = outputSpan.data();
+                UChar* outputEnd = std::to_address(outputSpan.end());
+                const char* input = byteCast<char>(icuInput.data());
+                const char* inputEnd = byteCast<char>(std::to_address(icuInput.end()));
+                ucnv_toUnicode(icuConverter.get(), &output, outputEnd, &input, inputEnd, nullptr, true, &error);
```

Comment:
I think it might be safer to have a single cover for `ucnv_toUnicode` called in all these places so the length handling and pointer stuff can be done right just once.